base/iterator.jl

```diff
@@ -63,13 +63,13 @@ function tuple_type_cons{S,T<:Tuple}(::Type{S}, ::Type{T})
     Tuple{S, T.parameters...}
 end
 eltype{I,Z}(::Type{Zip{I,Z}}) = tuple_type_cons(eltype(I), eltype(Z))
-start(z::Zip) = tuple(start(z.a), start(z.z))
-function next(z::Zip, st)
+@inline start(z::Zip) = tuple(start(z.a), start(z.z))
```

Comment:
Ah, I would have hoped that @timholy would fix this while he was at it. Too late. ;-)